src/pattern-library/components/Library.tsx

```diff
@@ -265,6 +265,86 @@ function Demo({
   );
 }
 
+export type LibraryExampleDemoProps = Omit<LibraryDemoProps, 'children'> & {
+  /**
+   * Example file to read and use as content (to be rendered with syntax
+   * highlighting)
+   */
+  exampleFile: string;
+};
+
+function ExampleDemo({
```

Comment:
This component duplicates a lot of code from the existing `Demo` one. The plan is to consolidate that as much as possible.